### PR TITLE
Smooth difficulty increase

### PR DESCRIPTION
**Pull request by Jontpan**
This commit will create a smooth difficulty curve for the player.
Also creates some changes to make adding new platforms easier.

#32

### CODE REVIEW

File: 2Dgame/Assets/Scripts/PlatformSpawnerScript.cs [hunk 1]
```diff
@@ -42,34 +42,61 @@ public class PlatformSpawnerScript : MonoBehaviour {
 	 * 		10% chance moving
 	 */ 
 	GameObject basicPlatformScheme() {
```
Comment (AxlLind) [inline 2Dgame/Assets/Scripts/PlatformSpawnerScript.cs:44]: Maybe just remove this?

File: 2Dgame/Assets/Scripts/PlatformSpawnerScript.cs [hunk 2]
```diff
@@ -42,34 +42,61 @@ public class PlatformSpawnerScript : MonoBehaviour {
...
+     * Is the game's primary progressionsystem.
+     * Difficulty increases as score increases.
      */
     GameObject choosePlatform()
     {
-        if (this.transform.position.y > 1800 && this.transform.position.y < 2000)
+        if (this.transform.position.y < 500)
         {
-            return movingPlatform;
+            return basicPlatformScheme();
```
Comment (AxlLind) [inline 2Dgame/Assets/Scripts/PlatformSpawnerScript.cs:79]: Change to 'return PlatformScheme(80, 10, 10)'

**Review by AxlLind — APPROVED** [review]
Really nice! We might want to tweak some of those ranges after we do some play-testing. Only thing I noticed is that the 'basicPlatformScheme()' is a bit redundant now.

Nice solution!

File: 2Dgame/Assets/Scripts/PlatformSpawnerScript.cs (hunk 2 above)
Comment (Jontpan) [reply 2Dgame/Assets/Scripts/PlatformSpawnerScript.cs:79]: My reasoning for keeping it that way was readability, and the stability of always knowing what our "basic" percentages are. Maybe even to the point of creating
`allmoving()
{
return PlatformScheme(0, 0, 100);
}`

and then calling that. Maybe that's overkill.

**Comment by Jontpan** [comment]
Didn't even know you could approve pull reqs. Sick.

Ranges should definitely be tweaked!

**Pull request merged**